pkg/httpserver/crud/errors.go

```diff
@@ -1,5 +1,75 @@
 package crud
 
-import "fmt"
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/justtrackio/gosoline/pkg/db"
+	"github.com/justtrackio/gosoline/pkg/db-repo"
+	"github.com/justtrackio/gosoline/pkg/httpserver"
+	"github.com/justtrackio/gosoline/pkg/log"
+	"github.com/justtrackio/gosoline/pkg/validation"
+)
 
 var ErrModelNotChanged = fmt.Errorf("nothing has changed on model")
+
+// handleErrorOnWrite handles errors for read operations.
+// Covers many default errors and responses like
+//   - context.Canceled, context.DeadlineExceed -> HTTP 299
+//   - validation.Error -> HTTP 400
+func handleErrorOnRead(logger log.Logger, err error) (*httpserver.Response, error) {
+	if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+		logger.Info("failed to read model(s): %s", err.Error())
+
+		return httpserver.NewStatusResponse(299), nil
+	}
+
+	if errors.As(err, &db_repo.RecordNotFoundError{}) {
+		logger.Warn("failed to read model: %s", err)
+
+		return httpserver.NewStatusResponse(http.StatusNotFound), nil
+	}
+
+	// rely on the outside handling of access forbidden and HTTP 500
+	return nil, err
+}
+
+// handleErrorOnWrite handles errors for write operations.
+// Covers many default errors and responses like
+//   - context.Canceled, context.DeadlineExceed -> HTTP 500
+//   - db_repo.RecordNotFoundError -> HTTP 404
+//   - ErrModelNotChanged -> HTTP 304
+//   - db.IsDuplicateEntryError -> HTTP 409
+//   - validation.Error -> HTTP 400
+func handleErrorOnWrite(logger log.Logger, err error) (*httpserver.Response, error) {
```

Comment:
Does it make sense to implement these methods on the base handler and then pass a context instead of a logger? Well, one would need to handle the cases where fields get set on the logger...